client/azure_client_set.go

```diff
@@ -89,89 +92,103 @@ func NewAzureClientSet(config AzureClientSetConfig) (*AzureClientSet, error) {
 func newDeploymentsClient(config *clientConfig) *resources.DeploymentsClient {
 	c := resources.NewDeploymentsClientWithBaseURI(config.resourceManagerEndpoint, config.subscriptionID)
 	c.Authorizer = autorest.NewBearerAuthorizer(config.servicePrincipalToken)
+	_ = c.AddToUserAgent(config.partnerIdUserAgent)
```

Comment:
What does this thing return?

Reply:
error when the passed parameter is empty, leaving the user-agent unchanged

Reply:
Then I would prefer to panic here.

Reply:
What do you think about checking if config.PartnerID contains something before creating `clientConfig` and panicking on that case? Otherwise I need to get the error returned by `AddToUserAgent` on every function that creates a client. It's more explicity, but also more repetitive.

Reply:
I pushed the explicity/repetitive version. Still, let me know what you think about failing earlier if `config.PartnerID` is not set.